### PR TITLE
Update TIDB specific function docs

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

1. Make sure the page TOC links to the functions 
2. Remove argument description from the page TOC
3. Add info for `VITESS_HASH()`
4. Reorganize the functions in alphabetical order

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

https://github.com/pingcap/tidb/issues/20971

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -3092,7 +3092,7 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Default value: `ON`
 - This variable controls whether to generate binary-encoded execution plans in slow logs and statement summaries.
 - When this variable is set to `ON`, you can view visual execution plans in TiDB Dashboard. Note that TiDB Dashboard only provides visual display for execution plans generated after this variable is enabled.
-- You can execute the `SELECT tidb_decode_binary_plan('xxx...')` statement to parse the specific plan from a binary plan.
+- You can execute the []`SELECT tidb_decode_binary_plan('xxx...')`](/functions-and-operators/tidb-functions.md#tidb_decode_binary_plan) statement to parse the specific plan from a binary plan.
```
Comment (lilin90) [inline system-variables.md:3095]:
```suggestion
- You can execute the [`SELECT tidb_decode_binary_plan('xxx...')`](/functions-and-operators/tidb-functions.md#tidb_decode_binary_plan) statement to parse the specific plan from a binary plan.
```

**Review by lilin90 — COMMENTED** [review]
I haven't finished my review and will finish it soon.

**Review by lilin90 — APPROVED** [review]
/lgtm

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -360,50 +481,96 @@ The output is as follows:
...
+Race Enabled: false
+TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
+Check Table Before Drop: false
 1 row in set (0.00 sec)
 ```
 
-Execute `SET RESOURCE GROUP` to set the resource group for the current session to `rg2`, and then view the resource group bound to the current user:
+## VITESS_HASH
+
+The `VITESS_HASH(num)` function is used to hash a number in the same way Vitess does this. This is to aid migration from Vitess to TiDB.
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:561]: > | `VITESS_HASH(str)` | The `VITESS_HASH` function returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |

@dveeden The previous document uses `VITESS_HASH(str)`. Here, it is `VITESS_HASH(num)`. Could you please confirm which one is more accurate? 


I tested this function in v8.1.0 and it seems that TiDB will attempt to truncate a string to a number. Should we describe this behavior here?


```shell
mysql> SELECT VITESS_HASH(123);
+---------------------+
| VITESS_HASH(123)    |
+---------------------+
| 1155070131015363447 |
+---------------------+
1 row in set (0.00 sec)

mysql> SELECT VITESS_HASH("123");
+---------------------+
| VITESS_HASH("123")  |
+---------------------+
| 1155070131015363447 |
+---------------------+
1 row in set (0.01 sec)

mysql> SELECT VITESS_HASH("123a");
+---------------------+
| VITESS_HASH("123a") |
+---------------------+
| 1155070131015363447 |
+---------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------+
| Level   | Code | Message                                   |
+---------+------+-------------------------------------------+
| Warning | 1292 | Truncated incorrect INTEGER value: '123a' |
+---------+------+-------------------------------------------+
1 row in set (0.00 sec)
```

Comment (dveeden) [reply functions-and-operators/tidb-functions.md:561]: `num` is more accurate.

The conversion, earning etc is generic behavior and not specific to this function.

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -11,47 +11,164 @@ The following functions are TiDB extensions, and are not present in MySQL:
...
+| [`TIDB_PARSE_TSO()`](#tidb_parse_tso) | Extracts the physical timestamp from a TiDB TSO timestamp. See also: [`tidb_current_ts`](/system-variables.md#tidb_current_ts). |
+| [`TIDB_PARSE_TSO_LOGICAL()`](#tidb_parse_tso_logical) | Extracts the logical timestamp from a TiDB TSO timestamp. |
+| [`TIDB_ROW_CHECKSUM()`](#tidb_row_checksum) | Queries the checksum value of a row. This function can only be used in `SELECT` statements within the FastPlan process. That is, you can query through statements like `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id = ?` or `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id IN (?, ?, ...)`. See also: [Data integrity validation for single-row data](/ticdc/ticdc-integrity-check.md). |
+| [`TIDB_SHARD()`](#tidb_shard) | Creates a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD` function as the prefix.|
+| [`TIDB_VERSION()`](#tidb_version) | Returns the TiDB version with additional build information. |
+| [`VITESS_HASH()`](#vitess_hash) | Returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
```
Comment (lilin90) [inline functions-and-operators/tidb-functions.md:28]:
```suggestion
| [`VITESS_HASH()`](#vitess_hash) | Returns the hash of a number that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
```

File: functions-and-operators/tidb-functions.md [hunk 4]
```diff
@@ -11,47 +11,164 @@ The following functions are TiDB extensions, and are not present in MySQL:
...
+| [`TIDB_PARSE_TSO()`](#tidb_parse_tso) | Extracts the physical timestamp from a TiDB TSO timestamp. See also: [`tidb_current_ts`](/system-variables.md#tidb_current_ts). |
+| [`TIDB_PARSE_TSO_LOGICAL()`](#tidb_parse_tso_logical) | Extracts the logical timestamp from a TiDB TSO timestamp. |
+| [`TIDB_ROW_CHECKSUM()`](#tidb_row_checksum) | Queries the checksum value of a row. This function can only be used in `SELECT` statements within the FastPlan process. That is, you can query through statements like `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id = ?` or `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id IN (?, ?, ...)`. See also: [Data integrity validation for single-row data](https://docs.pingcap.com/tidb/stable/ticdc-integrity-check). |
+| [`TIDB_SHARD()`](#tidb_shard) | Creates a shard index to scatter the index hotspot. A shard index is an expression index with a `TIDB_SHARD` function as the prefix.|
+| [`TIDB_VERSION()`](#tidb_version) | Returns the TiDB version with additional build information. |
+| [`VITESS_HASH()`](#vitess_hash) | Returns the hash of a string that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
```
Comment (lilin90) [inline functions-and-operators/tidb-functions.md:50]:
```suggestion
| [`VITESS_HASH()`](#vitess_hash) | Returns the hash of a number that is compatible with Vitess' `HASH` function. This is intended to help the data migration from Vitess. |
```

**Review by lilin90 — COMMENTED** [review]
@dveeden Then should we update the "string" to "number" for descriptions at the table?

**Pull request merged**